ipaplatform/debian/tasks.py

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright (C) 2017  FreeIPA Contributors see COPYING for license
+#
+
+"""
+This module contains default Debian-specific implementations of system tasks.
+"""
+
+from ipaplatform.paths import paths
+from ipaplatform.base.tasks import *
+from ipaplatform.redhat.tasks import RedHatTaskNamespace
+
+BaseTask = BaseTaskNamespace()
+
+class DebianTaskNamespace(RedHatTaskNamespace):
+
+    def restore_pre_ipa_client_configuration(self, fstore, statestore,
+                                             was_sssd_installed,
+                                             was_sssd_configured):
+        # Debian doesn't use authconfig, nothing to restore
+        return True
+
+    def set_nisdomain(self, nisdomain):
+        # Debian doesn't use authconfig, nothing to set
+        return True
+
+    def modify_nsswitch_pam_stack(self, sssd, mkhomedir, statestore):
+        # Debian doesn't use authconfig, this is handled by pam-auth-update
+        return True
+
+    def modify_pam_to_use_krb5(self, statestore):
+        # Debian doesn't use authconfig, this is handled by pam-auth-update
+        return True
+
+    def backup_auth_configuration(self, path):
+        # Debian doesn't use authconfig, nothing to backup
+        return True
+
+    def restore_auth_configuration(self, path):
+        # Debian doesn't use authconfig, nothing to restore
+        return True
+
+    def parse_ipa_version(self, version):
+        return BaseTask.parse_ipa_version(version)
```

Comment:
What I meant was to add `@staticmethod` before BaseTaskNamespace parse_version definition so that it goes something like this:
```python
class BaseTaskNamespace(object):
    ...
    @staticmethod
        def parse_ipa_version(version):
        """
        :param version: textual version
        :return: object implementing proper __cmp__ method for version compare
        """
        return parse_version(version)
```
Note the removal of `self` which is not needed in the method anyway so it being static is actually the preferred way. Once you do this, you can go:
```python
class DebianTaskNamespace(RedHatTaskNamespace):
    ...
    def parse_ipa_version(self, version):
        return BaseTaskNamespace.parse_ipa_version(version)
```
Note that your `parse_ipa_version` can itself be static as it uses none of the instance members stored in `self` :)